### PR TITLE
Remove invalid `timeout` argument for dist.barrier

### LINKED ISSUE #4480: there is not keyword argument 'timeout' in torch.distributed.barrier()
**Describe the bug**
there is not keyword argument 'timeout' in `torch.distributed.barrier()`'s arguments 

https://github.com/NVIDIA/Megatron-LM/blob/03f4111bf8b67a76beae3cfa7baa9dc8f2d3eef3/megatron/training/global_vars.py#L105
https://github.com/NVIDIA/Megatron-LM/blob/03f4111bf8b67a76beae3cfa7baa9dc8f2d3eef3/tests/unit_tests/test_utilities.py#L106

Image

A clear and concise description of what the bug is. Tag the [@mcore-oncall](https://github.com/orgs/NVIDIA/teams/mcore-oncall) 
to get oncall's attention to this issue.

**Steps/Code to reproduce bug**

Please list *minimal* steps or code snippet for us to be able to reproduce the bug.

A helpful guide on on how to craft a minimal bug report http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports.

```
torchrun --nproc_per_node=8 -m pytest -v tests/unit_tests/test_parallel_state.py -p no:randomly
```

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by zhaoyinglia**
# What does this PR do ?

Remove invalid parameters for torch.distributed.barrier()

:warning: For major changes (either in lines of code or in its impact), please make sure to first share a design doc with the team. If you're unsure what's the best way to do so, contact the @mcore-oncall.

## Issue tracking

For PRs from open-source community contributors:
- **Small updates (bug fixes, minor improvements)**: a linked issue is **recommended** and will accelerate the PR review process.

Linked issue: Related to #4480 

### Pre-checks

- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

Feel free to message or comment the [@mcore-oncall](https://github.com/orgs/NVIDIA/teams/mcore-oncall) to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

All PRs start as **draft**. If you open a non-draft PR, it will be automatically converted to draft.

#### Step 1: Mark PR as "Ready for Review"

1. When your PR is ready, click **Ready for Review**.
2. An oncall reviewer is auto-assigned and expert reviewers are notified based on your changes.
   - Some PRs may jump straight to step 2. This is determined by `.github/CODEOWNERS`.

:warning: Only mark as ready once merge-conflicts are resolved and the CI is passing.
Final Review might get declined if these requirements are not fulfilled.

#### Step 2: Final Review

For PRs that change `megatron/core`, once all expert reviewers have approved, the `Final Review` label is applied **automatically** and final reviewers are assigned.

For PRs outside `megatron/core`, this step is skipped.

#### Step 3: Approved

Once all required reviewers have approved, the `Approved` label is applied **automatically**.

### Merge

Any member of [mcore-engineers](https://github.com/orgs/NVIDIA/teams/mcore-engineers) will be able to merge your PR.

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### CODE REVIEW

**Review by maanug-nv — APPROVED** [review]
lgtm, thanks for the fix

**Review by ko3n1g — CHANGES_REQUESTED** [review]
We added this to ensure we don’t hang on failing tests for too long. Is this still guaranteed if we remove the timeout?

**Comment by maanug-nv** [comment]
> We added this to ensure we don’t hang on failing tests for too long. Is this still guaranteed if we remove the timeout?

'timeout' is not an argument for `torch.distributed.barrier()`: https://docs.pytorch.org/docs/2.11/distributed.html#torch.distributed.barrier

Perhaps the barriers in unit tests should be switched to `torch.distributed.monitored_barrier()`?

**Review by deepakn94 — APPROVED** [review]

**Review by ko3n1g — APPROVED** [review]

**Comment by asolergi-nv** [comment]
/ok to test [e8d6619](https://github.com/NVIDIA/Megatron-LM/pull/4512/commits/e8d661930bd000b116731b21ea6702253f5638d3)
